server/lobbyconnection.py

```diff
@@ -56,6 +56,31 @@ def __init__(self, message, recoverable=True, *args, **kwargs):
         self.recoverable = recoverable
 
 
+class BanError(Exception):
+    """
+    Represents a BanError
+
+    It is expected that the connection be terminated immediately.
+    """
+    def __init__(self, ban_expiry, ban_reason, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.ban_expiry = ban_expiry
+        self.ban_reason = ban_reason
+
+    def message(self):
+        return (f"You are banned from FAF {self._ban_duration_text()}. <br>"
+                f"Reason : <br>"
+                f"{self.ban_reason}")
+
+    def _ban_duration_text(self):
+        ban_duration = self.ban_expiry - datetime.now()
+        if ban_duration.days > 365 * 100:
+            return "forever"
+        humanized_ban_duration = humanize.precisedelta(ban_duration, minimum_unit='hours')
```

Comment:
Right now the code is still pretty inconsistent on quote usage, but I've been trying to convert everything over to double quotes (unless the string contains a double quote).
```suggestion
        humanized_ban_duration = humanize.precisedelta(ban_duration, minimum_unit="hours")
```

Reply:
I really think this should be done later as a big PR on the whole codebase rather than trying to get it right here.

Reply:
(Am volunteering.)

Reply:
The tricky thing about doing it all at once is that it could cause conflicts with other PR's. If you want to take a stab at that, go ahead, maybe it won't be too bad. But that still does't justify committing new code with single quotes ;)

Reply:
It's even coding style bullet number 1 as listed in the contributing document: [CONTRIBUTING.md](https://github.com/FAForever/server/blob/develop/CONTRIBUTING.md)

Reply:
cool, pushed again. opened a PR for mass-replacing too to see if it's workable: #648.